### PR TITLE
Ignore but log Live L2 txs parse errors

### PR DESCRIPTION
**Pull request by adamiak**

### CODE REVIEW

File: packages/backend/src/peripherals/starkware/schema/PerpetualLiveL2TransactionResponse.ts [hunk 1]
```diff
@@ -5,14 +5,19 @@ import { PerpetualL2Transaction } from './PerpetualBatchInfoResponse'
 const PerpetualLiveL2TransactionResponseTransactionInfo = z.object({
   tx: PerpetualL2Transaction,
   tx_id: z.number(),
+  parseError: z.undefined(),
```
Comment (torztomasz) [inline packages/backend/src/peripherals/starkware/schema/PerpetualLiveL2TransactionResponse.ts:8]: why is that z.undefined?

**Review by torztomasz — DISMISSED** [review]

File: packages/backend/src/peripherals/starkware/schema/PerpetualLiveL2TransactionResponse.ts (hunk 1 above)
Comment (adamiak) [reply packages/backend/src/peripherals/starkware/schema/PerpetualLiveL2TransactionResponse.ts]: Removed here and added as a separate discriminated union called ParsedL2TransactionInfo

File: packages/backend/src/core/sync/LiveL2TransactionDownloader.test.ts [hunk 2]
```diff
@@ -86,17 +87,19 @@ describe(LiveL2TransactionDownloader.name, () => {
...
-          ).toHaveBeenNthCalledWith(
-            i + 1,
-            {
-              transactionId: tx.transactionId,
-              timestamp: tx.timestamp,
-              data: tx.transaction,
-            },
-            mockKnexTransaction
-          )
+          if (tx.parsedSuccessfully) {
```
Comment (michalsidzej) [inline packages/backend/src/core/sync/LiveL2TransactionDownloader.test.ts:90]: I'd suggest adding a test that checks what happens if the transaction is not parsed successfully.

**Review by michalsidzej — APPROVED** [review]
This change is quite simple, but touches quite a few logical checks. Would be great to unit test a less happy path. This is a nit pick, but otherwise LGTM

File: packages/backend/src/core/sync/LiveL2TransactionDownloader.test.ts (hunk 2 above)
Comment (adamiak) [reply packages/backend/src/core/sync/LiveL2TransactionDownloader.test.ts:132]: Added

**Pull request merged**